### PR TITLE
Setup environment, database, and created prisma object models

### PR DESCRIPTION
**Pull request by Brianna-Flo**
this commit will add initial prisma schemas for a user, recipe, ingredient, and grocery list

## Description

<what this pull request is doing>
Populated repository with initial frontend and backend folders
Setup and connected database using pgAdmin
Created prisma schema models for a user, ingredient, recipe, and grocery list
<what will be done in later PRs and not included here>
Creating routing endpoints to populate the database with information for each model object

## Milestones
<the milestones or stories/features that this works towards>
Setting up the backend database with object models

## Resources
<links to tutorials, code snippets, inspirations>
<if you took or translated specific code from the internet, please call it out here!>
Prisma documentation on model relations was used
https://www.prisma.io/docs/orm/prisma-schema/data-model/relations/one-to-one-relations
https://www.prisma.io/docs/orm/prisma-schema/data-model/relations/many-to-many-relations

## Test Plan
<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>

### CODE REVIEW

File: backend/prisma/schema.prisma [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+model Ingredient {
+  id             Int          @id @default(autoincrement())
+  ingredientName String
+  quantity       Int
+  aisle          String?
+  productImage   String?
+  expiration     DateTime?
+  estimatedCost  Decimal?
+  userId         Int // if ingredient is an ingredient a user owns
```
Comment (paulshaoyuqiao) [inline backend/prisma/schema.prisma:33]: are we sure we want a foreign key from an ingredient referencing a user? This could result in a lot of ingredient rows because common ingredients will likely be used in most recipes and subsequently most users.

File: backend/prisma/schema.prisma [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  servings                           Int
+  dietaryPreferencesAndIntollerences String?
+  ingredientsNeeded                  Json[] // from spoonacular api
+  sourceUrl                          String
+  totalCost                          Decimal // total cost of ingredients to create
+  instructions                       Json[] // store json return of steps
+}
+
+model GroceryList {
+  id     Int          @id @default(autoincrement())
```
Comment (paulshaoyuqiao) [inline backend/prisma/schema.prisma:55]: we might want fields also for:

- name of grocery list
- which week the grocery list is for (some kind of date range)

File: backend/prisma/schema.prisma [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+model User {
+  id                Int          @id @default(autoincrement())
+  username          String       @unique
+  password          String
+  ingredientsOnHand Ingredient[]
+  groceryList       GroceryList? // store the grocery list for a user
+}
+
+model Ingredient {
+  id             Int          @id @default(autoincrement())
```
Comment (paulshaoyuqiao) [inline backend/prisma/schema.prisma:26]: we might want unit cost for each ingredient as well, so the overall recipe could just take into account the amount of ingredients as part of calculation?

File: backend/prisma/schema.prisma [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+// many to many relationship
+// https://www.prisma.io/docs/orm/prisma-schema/data-model/relations/many-to-many-relations
+// recipe can be stored in what the user will make throughout the week, their "to make" list
+model Recipe {
+  id                                 Int     @id @default(autoincrement()) // store the id number from spoonacular api, otherwise id is auto incremented (part of the users database)
+  recipeTitle                        String
+  previewImage                       String? // if image not provided, generic image will be presented
+  servings                           Int
+  dietaryPreferencesAndIntollerences String?
```
Comment (paulshaoyuqiao) [inline backend/prisma/schema.prisma:47]: might be better to separate these out instead into different boolean fields so it will be much easier and cleaner for you to check later on:

- containsDiary
- containsNuts
etc.

File: backend/prisma/schema.prisma [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+  ingredientsNeeded                  Json[] // from spoonacular api
```
Comment (paulshaoyuqiao) [inline backend/prisma/schema.prisma:48]: I think this field should be much simpler, referencing your ingredient model defined above. 
in general, we don't store the full json in a data model since it is not space efficient and also hard to read, by the time the data are stored in a table, we expect them to be clean and concise, otherwise, the business logic we later need to perform (such as filtering for certain recipes) will become less readable and more complex as well.

File: backend/prisma/schema.prisma (hunk 1 above)
Comment (Brianna-Flo) [reply backend/prisma/schema.prisma:33]: Ah I see that could result in a lot of information in the database. The idea I had in mind would be each ingredient would map to a user since the ingredient would have user specific information such as expiration date and quantity (example: I have a different expiration date of milk than another user), might there be another option or database to use that would handle this more efficiently?

File: backend/prisma/schema.prisma (hunk 5 above)
Comment (Brianna-Flo) [reply backend/prisma/schema.prisma:48]: For the ingredients needed, I was considering storing it in an array of Ingredient objects, however was unsure whether this would be the best option due to your comment earlier regarding having keys that reference other objects resulting in many ingredients?

File: backend/prisma/schema.prisma (hunk 2 above)
Comment (Brianna-Flo) [reply backend/prisma/schema.prisma:55]: Added!

File: backend/prisma/schema.prisma (hunk 3 above)
Comment (Brianna-Flo) [reply backend/prisma/schema.prisma:26]: Added!
